### PR TITLE
Clarify use of null in lattice_vectors

### LINKED ISSUE #196: Format of 'lattice_vectors' is unclear for 2D, 1D and 0D systems
Specification defines `lattice_vectors` as three lattice vectors in Cartesian coordinates. However, it is not clear how the vectors are defined for nonperiodic dimensions. Currently it seems that such vectors are ignored, and as such can contain any values. But it would probably make sense to have a default, and I would suggest having `null`s for all the coordinates of vectors in nonperiodic dimensions.

An example of a 2D surface on XZ plane:
```
...
  "dimension_types": [1, 0, 1],
  "lattice_vectors": [[4, 0, 0], [null, null, null], [0, 0, 4]],
...
```

### PR DESCRIPTION
**Pull request by rartino**
Fixes #196

### CODE REVIEW

File: optimade.rst [hunk 1]
```diff
@@ -1743,7 +1743,8 @@ lattice\_vectors
     (Therefore, the first index runs over the three lattice vectors and the second index runs over the x, y, z Cartesian coordinates).
   - For databases that do not define an absolute Cartesian system (e.g., only defining the length and angles between vectors), the first lattice vector SHOULD be set along *x* and the second on the *xy*-plane.
   - This property MUST be an array of dimensions 3 times 3 regardless of the elements of property `dimension_types`_. The vectors SHOULD by convention be chosen so the determinant of the :property:`lattice_vectors` matrix is different from zero. The vectors in the non-periodic directions have no significance beyond fulfilling these requirements.
```
Comment (ml-evs) [inline optimade.rst]: Just a minor suggestion for consistency in this block.

```suggestion
   - This property MUST be an array of dimensions 3 times 3 regardless of the elements of :property:`dimension\_types`. The vectors SHOULD by convention be chosen so the determinant of the :property:`lattice_vectors` matrix is different from zero. The vectors in the non-periodic directions have no significance beyond fulfilling these requirements.
```

File: optimade.rst [hunk 2]
```diff
@@ -1743,7 +1743,8 @@ lattice\_vectors
...
-    
+  - All three elements of the inner lists of floats MAY be :val:`null` for nonperiodic dimensions, i.e., those dimensions for which :property:`dimension\_types` is 0.
```
Comment (ml-evs) [inline optimade.rst]: Elsewhere in the spec we've hyphenated "non-periodic":

```suggestion
  - All three elements of the inner lists of floats MAY be :val:`null` for non-periodic dimensions, i.e., those dimensions for which :property:`dimension\_types` is 0.
```

File: optimade.rst [hunk 3]
```diff
@@ -1742,8 +1742,9 @@ lattice\_vectors
...
     (Therefore, the first index runs over the three lattice vectors and the second index runs over the x, y, z Cartesian coordinates).
   - For databases that do not define an absolute Cartesian system (e.g., only defining the length and angles between vectors), the first lattice vector SHOULD be set along *x* and the second on the *xy*-plane.
-  - This property MUST be an array of dimensions 3 times 3 regardless of the elements of property `dimension_types`_. The vectors SHOULD by convention be chosen so the determinant of the :property:`lattice_vectors` matrix is different from zero. The vectors in the non-periodic directions have no significance beyond fulfilling these requirements.
-    
+   - This property MUST be an array of dimensions 3 times 3 regardless of the elements of :property:`dimension\_types`. The vectors SHOULD by convention be chosen so the determinant of the :property:`lattice_vectors` matrix is different from zero. The vectors in the non-periodic directions have no significance beyond fulfilling these requirements.
```
Comment (CasperWA) [inline optimade.rst]:
```suggestion
   - This property MUST be an array of dimensions 3 times 3 regardless of the elements of :property:`dimension\_types`.
     The vectors SHOULD by convention be chosen so the determinant of the :property:`lattice_vectors` matrix is different from zero.
     The vectors in the non-periodic directions have no significance beyond fulfilling these requirements.
```

File: optimade.rst [hunk 4]
```diff
@@ -1742,8 +1742,9 @@ lattice\_vectors
...
+  - All three elements of the inner lists of floats MAY be :val:`null` for non-periodic dimensions, i.e., those dimensions for which :property:`dimension\_types` is 0.
```
Comment (CasperWA) [inline optimade.rst]:
```suggestion
  - All three elements of the inner lists of floats MAY be :val:`null` for non-periodic dimensions, i.e., those dimensions for which :property:`dimension\_types` is :val:`0`.
```

File: optimade.rst [hunk 5]
```diff
@@ -1742,8 +1742,11 @@ lattice\_vectors
   - MUST be a list of three vectors *a*, *b*, and *c*, where each of the vectors MUST BE a list of the vector's coordinates along the x, y, and z Cartesian coordinates.
     (Therefore, the first index runs over the three lattice vectors and the second index runs over the x, y, z Cartesian coordinates).
   - For databases that do not define an absolute Cartesian system (e.g., only defining the length and angles between vectors), the first lattice vector SHOULD be set along *x* and the second on the *xy*-plane.
-  - This property MUST be an array of dimensions 3 times 3 regardless of the elements of property `dimension_types`_. The vectors SHOULD by convention be chosen so the determinant of the :property:`lattice_vectors` matrix is different from zero. The vectors in the non-periodic directions have no significance beyond fulfilling these requirements.
-    
+   - This property MUST be an array of dimensions 3 times 3 regardless of the elements of :property:`dimension\_types`.
```
Comment (CasperWA) [inline optimade.rst]:
```suggestion
  - This property MUST be an array of dimensions 3 times 3 regardless of the elements of :property:`dimension\_types`.
```

File: optimade.rst [hunk 6]
```diff
@@ -1742,8 +1742,11 @@ lattice\_vectors
...
+     The vectors SHOULD by convention be chosen so the determinant of the :property:`lattice_vectors` matrix is different from zero.
```
Comment (CasperWA) [inline optimade.rst]:
```suggestion
    The vectors SHOULD by convention be chosen so the determinant of the :property:`lattice_vectors` matrix is different from zero.
```

File: optimade.rst [hunk 7]
```diff
@@ -1742,8 +1742,11 @@ lattice\_vectors
...
+     The vectors in the non-periodic directions have no significance beyond fulfilling these requirements.
```
Comment (CasperWA) [inline optimade.rst]:
```suggestion
    The vectors in the non-periodic directions have no significance beyond fulfilling these requirements.
```

**Comment by rartino** [comment]
Weird indentation error, well, it should be fixed now.

**Pull request merged**